types/react-native/index.d.ts

```diff
@@ -8634,6 +8638,11 @@ export interface SwitchProps extends SwitchPropsIOS {
      */
     disabled?: boolean;
 
+    /**
+     * Invoked with the the change event as an argument when the value changes.
+     */
+    onChange?: ((event?: SwitchChangeEvent) => Promise<void> | void) | null;
```

Comment:
I'm not familiar with this API, but it looks like `onValueChange`, below, can't be promise-returning or `null`.  Is the mismatch intentional?

Reply:
Incidentally, you should be able to assign a promise-returning function to a void-returning function type, so I would expect the tests to pass without the `Promise<void> |`.

Reply:
As for `null`, I'll let you decide, but the two should probably be consistent.

Reply:
I added `Promise` return type as used in source code  https://github.com/facebook/react-native/blob/master/Libraries/Components/Switch/Switch.js#L79 

I checked locally, all tests are passed without `Promise<void> |`. So I will remove it if you think it's for the best

As for `null`, @ca057 said me to add `null` check in the conversation above

Reply:
I don't feel strongly about either change, but I think the two signatures should agree.

Reply:
> I think the two signatures should agree.
Good point. I think we should follow the typings in the React Native codebase, as out of my perspective this should be the source of truth which is the most likely to be correct. I would therefore propose to adapt the typing of `onValueChange` as well to properly reflect the maybe-type and the possible return values. What do you think?

Maybe @alloy can comment how these adaptions were done previously in the typings?

Reply:
@ca057 I totally agree with you that react native codebase should be source of truth for typing.

Should I change `onValueChange ` method signature as you described above? Or will we be waiting for @alloy answer?

Reply:
I would align it and fix the `onValueChange` as well according to the types in the React Native repo.

Reply:
@ca057  `onValueChange` method has been updated

Reply:
Ci has found type error in react-dialog lib after my changes in method `onValueChange`, should I change this lib types definition too? I'm afraid  it might will broken something else.

Reply:
I think we should fix the types there too. Can you do it or shall I help to get this type update done? :)

Reply:
Woah, this type is weird. I have no idea why it is Promise<void> in the main repo. It should just be `mixed` in Flow. It seems reasonable to align with Flow in this PR, but I think the upstream code is wrong too.

Reply:
@amcasey If you don't have any more questions, could you resolve your "changes requested" status please

Reply:
Sorry about that.  Looks like you managed to merge anyway though. :smile: